frameworklauncher/src/main/java/com/microsoft/frameworklauncher/applicationmaster/Node.java

```diff
@@ -48,31 +48,25 @@ public static Node fromNodeReport(NodeReport nodeReport) throws Exception {
         ResourceDescriptor.fromResource(nodeReport.getUsed()));
   }
 
-  // Compare two node's AvailableResource,  order is Gpu, Cpu, Memory
+  // Compare two node gpu resource.
+  // Fist compare two nodes total Gpu resource,
+  // Then compare two nodes available Gpu resource.
   @Override
   public int compareTo(Node other) {
     if (other == null) {
       return -1;
     }
 
-    ResourceDescriptor thisAvailableResource = this.getAvailableResource();
-    ResourceDescriptor otherAvailableResource = other.getAvailableResource();
```

Comment:
Please first extract these var, otherwise the AvailableResource will compute again and again.